File: Bot/commands/bind.js
const {SlashCommandBuilder} = require("@discordjs/builders");
const {addSubcommands, saveExecute} = require("../utils/utilities");

module.exports = {
    data: new SlashCommandBuilder()
        .setName('bind')
        .setDescription('Binds a roleplay character to an entity (army, trader etc.)')
        .addSubcommand(subcommand =>
            subcommand
                .setName('army-or-company')
                .setDescription('Binds a character to an army or trading/armed company')
                .addStringOption(option =>
                    option.setName('army-or-company-name')
                        .setDescription('The name of the army/company')
                        .setRequired(true))
                .addUserOption(option =>
                    option.setName("target-player")
                        .setDescription("Player that gets bound to the army, PING that discord account!")
                        .setRequired(true))
        ),
    async execute(interaction) {
        // Dynamically get all subcommands for called command
        const commands = addSubcommands('bind', false);
        const toExecute = commands[interaction.options.getSubcommand()];
        saveExecute(toExecute, interaction);
    },
};

File: ALSpringBackend/src/main/java/com/ardaslegends/data/presentation/discord/commands/bind/Bind.java
package com.ardaslegends.data.presentation.discord.commands.bind;

import com.ardaslegends.data.presentation.discord.commands.ALCommand;
import com.ardaslegends.data.presentation.discord.commands.ALCommandExecutor;
import com.ardaslegends.data.service.ArmyService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.javacord.api.DiscordApi;
import org.javacord.api.entity.message.embed.EmbedBuilder;
import org.javacord.api.interaction.*;
import org.springframework.stereotype.Component;

import java.util.Arrays;
import java.util.Map;

@RequiredArgsConstructor

@Slf4j
@Component
public class Bind implements ALCommand {

    private final DiscordApi api;

    private final ArmyService armyService;

    public SlashCommand bind;

    public void init(Map<String, ALCommandExecutor> commands) {
        log.debug("Initializing /bind command");
        this.bind = SlashCommand.with("bind", "Binds an entity (army/company) to a player", Arrays.asList(
                                new SlashCommandOptionBuilder()
                                        .setType(SlashCommandOptionType.SUB_COMMAND)
                                        .setName("army-or-company")
                                        .setDescription("JAVACORD Binds a character to an army or trading/armed company")
                                        .addOption(SlashCommandOption.createStringOption("army-or-company-name", "The name of the army/company", true))
                                        .addOption(SlashCommandOption.createUserOption("target-player", "Player that gets bound to the army, PING that discord account!", true))
                                        .build()
                        )
                )
                .createGlobal(api)
                .join();
        commands.put("bind army-or-company", new BindArmyOrCompanyCommand()::execute);
        log.info("Finished initializing /bind command");
    }

}
